src/components/cladeFilterDrawer/caseDefinitionConstructor/index.tsx

```diff
@@ -13,11 +13,18 @@ import {
 } from "../../../utils/caseDefinitionFilters";
 import { get_leaves } from "../../../utils/treeMethods";
 
-export const CaseDefinitionConstructor = () => {
+type CaseDefinitionConstructorProps = {
+  setCaseDefOpen: Function;
+};
+
+export const CaseDefinitionConstructor = (
+  props: CaseDefinitionConstructorProps
+) => {
   // @ts-ignore -- TODO: figure out how to add types to state
   const state = useSelector((state) => state.global);
   const dispatch = useDispatch();
   const [selectedFields, setSelectedFields] = useState<string[]>([]);
+  const { setCaseDefOpen } = props;

```

Comment:
This is reasonable, but something to think about is you might not want to pass the entire state setter down to this child component. CaseDefinitionConstructor never needs to be able to open the CaseDef, because it is itself only present when it's open (unless I'm misunderstanding something). Instead, you could pass in a prop called something like `closeCaseDef` (with signature `() => void`) and it would just be a simple function that, when called, causes the case definition to close. That way CaseDefinitionConstructor does not need to be aware of any implementation details.

And above, the parent component could create this function by "baking in" the state setter. Probably this:
```
const closeCaseDef = () => {
  setCaseDefOpen(false);
};
```

Then the above can be passed as a prop to the child and the child has the implementation details abstracted away. That said, this is a really simple case, it's just a boolean, so it's very unlikely that `setCaseDefOpen` would be misused here. Also you might already be aware of all this and chosen the expedient path of just doing `setCaseDefOpen(false)`. But I wanted to point out this approach in case you hadn't thought about it: for more complicated setters or where you need to "bake in" (AKA, closure) some kind of information available at the parent that the child doesn't really need, it's a useful trick.

Reply:
Thank you! Would love a mini explanation at some point of `closure`s and `promise`s. I think I already use both of those things, but would be good to get a little crisper on the actual definition / use case for them.